src/core/geometry/geom3/canonicalize.js

```diff
@@ -1,38 +1,37 @@
-const FuzzyFactory3d = require('../FuzzyFactory3d')
+const FuzzyFactory3 = require('../fuzzyfactory/FuzzyFactory3')
+
 const fromPolygons = require('./fromPolygons')
 
 /**
-   * Returns a canonicalized version of the input geometry : ie every very close
-   * points get deduplicated
-   * @returns {Geom3}
-   * @example
-   * let rawGeometry = someGeometryMakingFunction()
-   * let canonicalizedCSG = canonicalize(rawGeometry)
-   */
-const canonicalize = (geometry, options) => {
+ * Returns a canonicalized version of the input geometry : ie every very close
+ * points get deduplicated
+ * @returns {Geom3}
+ * @example
+ * let rawGeometry = someGeometryMakingFunction()
+ * let canonicalizedGeom = canonicalize(rawGeometry)
+ */
+const canonicalize = (geometry) => {
   if (geometry.isCanonicalized) {
     return geometry
-  } else {
-    const factory = new FuzzyFactory3d()
-    let result = Geom3FromFuzzyFactory(factory, geometry)
-    result.isCanonicalized = true
-    result.isRetesselated = geometry.isRetesselated
-    result.properties = geometry.properties // keep original properties
-    return result
   }
-}
 
-const Geom3FromFuzzyFactory = (factory, sourcecsg) => {
-  let _this = factory
-  let newpolygons = []
-  sourcecsg.polygons.forEach(function (polygon) {
-    let newpolygon = _this.getPolygon(polygon)
-    // see getPolygon above: we may get a polygon with no vertices, discard it:
+  const factory = new FuzzyFactory3()
+
+  const newpolygons = []
+  geometry.polygons.forEach((polygon) => {
+    const newpolygon = factory.getPolygon(polygon)
+
+    // discard incomplete polygons
     if (newpolygon.vertices.length >= 3) {
```

Comment:
should this discarding not take place in the factory itself ? this seems weirdly redundant ?
I would expect the factory to return valid things or at least pre-filtered so that we would not need to re-iterate here ?

Reply:
Added note to #139